src/cc/export/helpers.h

```diff
@@ -751,6 +753,27 @@ int bpf_l4_csum_replace_(void *ctx, u64 off, u64 from, u64 to, u64 flags) {
   return bpf_l4_csum_replace(ctx, off, from, to, flags);
 }
 
+static inline __attribute__((always_inline))
+BCC_SEC("helpers")
+const char *fd2path(unsigned int fd) {
+  struct files_struct *files = NULL;
+  struct fdtable *fdt = NULL;
+  struct file **fds = NULL;
+  struct file *f = NULL;
+  struct dentry *de = NULL;
+  struct qstr dn = {};
+  struct task_struct *curr = (struct task_struct *)bpf_get_current_task();
+
+  bpf_probe_read(&files, sizeof(files), &curr->files);
+  bpf_probe_read(&fdt, sizeof(fdt), &files->fdt);
```

Comment:
If bpf_probe_read() has issues (e.g., page fault) to read the contents, it will zero'ing the buffer and returns a negative value. With address of zero, all subsequent bpf_probe_read() will return -EFAULT, so eventually, dn.name will 0. I guess it is okay not to check bpf_probe_read() return value since we optimize for common case where bpf_probe_read() unlikely to fail. But could you add comments about why this mechanism works.